src/LibYear.Lib/FileTypes/XmlProject.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using NuGet.Versioning;
+
+namespace LibYear.Lib.FileTypes
+{
+    public abstract class XmlProject : IProjectFile
+    {
+        public virtual string FileName => throw new NotSupportedException();

```

Comment:
Why is this property `virtual`? Later on, you just override it, in your concrete class implementation but .. don't add any value to it?